### PR TITLE
Add citations for SCSB items

### PR DESCRIPTION
**Pull request by maxkadel**
I think this PR delivers value, but there are some things I'd like help improving.

Connected to #4012

- Right now, it is only tested for Books, I think we should add tests for other types of items, especially journals.
- It requires a second call to Solr, which seems like it shouldn't be necessary, but I've spent a lot of time in the catalog controller trying to add the `_citation_` fields so they are in the solr document at this point, but haven't succeeded yet. (See https://github.com/pulibrary/orangelight/issues/4429 for follow-up ticket to resolve this)
- ~Probably a lot of the methods in Blacklight::Document::Mla can be a) Private and b) moved into an abstract method for other CiteProc citations (APA, Chicago)~ - done
- Right now we have to take apart existing Solr fields to get the publication information - see https://github.com/pulibrary/bibdata/issues/2515 for follow-up ticket to add separate Solr fields so we do not have to process them here but can use them directly from Solr.
- ~For some reason I can't get the author name to show up correctly with the comma between the family name and the given name. I've added a pending test for this issue.~ (I'm wondering if this is because of the combining accent in the example? - it does work correctly in development for SCSB-8953469) - I have created a fork that has a test, this seems to only affect names ending in decomposed accents. This is a bug in CiteProc - see https://github.com/pulibrary/citeproc/tree/names_with_final_accents

image

### CODE REVIEW

File: app/components/orangelight/document/citation_component.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class Orangelight::Document::CitationComponent < Blacklight::Document::CitationComponent
+  DEFAULT_FORMATS = {
+    'blacklight.citation.mla': :export_as_mla,
```
[the remark this answers is not in the document]
Comment (maxkadel) [reply app/components/orangelight/document/citation_component.rb:5]: Ticket to route all types of records through the same routines - https://github.com/pulibrary/orangelight/issues/4428

File: spec/models/apa_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Blacklight::Document::Apa, citation: true do
```
Comment (sandbergja) [inline spec/models/apa_spec.rb:5]: Woohoo citation tag!

File: app/models/concerns/blacklight/document/apa.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Creates an html APA citation for non-Marc records
+module Blacklight::Document::Apa
```
Comment (sandbergja) [inline app/models/concerns/blacklight/document/apa.rb:4]: I like this structure with the 3 modules for the 3 citation styles

**Review by sandbergja — CHANGES_REQUESTED** [review]
Thanks, @maxkadel !  🦄  I saw one very small thing in the Gemfile, the actual code looks great to me.

**Review by sandbergja — APPROVED** [review]
Woohoo!  Thanks, @maxkadel 🦄

**Pull request merged**